platform/admin/configurator/admin-configurator-impl/src/main/java/org/codice/ddf/admin/configurator/impl/BundleOperation.java

```diff
@@ -129,4 +155,36 @@ private BundleStateService getBundleStateService() {
     private boolean lookupBundleState() {
         return getBundleStateService().getState(bundle) == BundleState.Active;
     }
+
+    private boolean isPermittedToViewBundle(Bundle bundle) {
+        ServiceReference<?>[] services = bundle.getRegisteredServices();
+        if (services == null || services.length == 0) {
+            return true;
+        }
+
+        for (ServiceReference<?> service : services) {
+            Object pidProperty = service.getProperty("service.pid");
+            if (pidProperty instanceof String) {
+                if (!isPermittedToViewService((String) pidProperty)) {
+                    return false;
+                }
+            }
+        }
+
+        return true;
+    }
+
+    private boolean isPermittedToViewService(String servicePid) {
+        KeyValueCollectionPermission serviceToCheck = new KeyValueCollectionPermission(
+                "view-service.pid",
+                new KeyValuePermission("service.pid", Sets.newHashSet(servicePid)));
+
+        try {
+            return security.runWithSubjectOrElevate(() -> SecurityUtils.getSubject()
```

Comment:
You should only need to call `runWithSubjectOrElevate` if this code is called from the karaf command line. Is that the case here? Otherwise you should just check for a Subject and if != null, check permissions, otherwise return false.

Reply:
I'll update and push shortly...hold off reviewing more.